src/test/java/io/stargate/sgv2/jsonapi/service/operation/model/command/clause/update/PushOperationTest.java

```diff
@@ -83,6 +83,26 @@ public void testPushToNonExistingNested() {
                                       """);
       assertThat(doc).isEqualTo(expected);
     }
+
+    // Test to ensure $push operations are done in alphabetic order by field
+    @Test
+    public void testPushToNonExistingOrdered() {
+      // Put targets in "wrong" order (different from expected execution order)
+      UpdateOperation oper =
+          UpdateOperator.PUSH.resolveOperation(
+              objectFromJson("{ \"subdoc.newArray\" : \"value\", \"array\": 3 }"));
+      assertThat(oper).isInstanceOf(PushOperation.class);
+      ObjectNode doc = objectFromJson("{ }");
+      assertThat(oper.updateDocument(doc, targetLocator)).isTrue();
+      ObjectNode expected =
+          objectFromJson(
+              """
+                      { "array": [ 3 ], "subdoc" : { "newArray": [ "value" ] } }
```

Comment:
Looking at this, it does seem a little strange that the keys are created in alphabetical order versus the order in which they were specified in the `$push` operation. I don't think it would cause any concrete issues in Mongoose, but just bringing this up for potential further discussion. Is there a reason to prefer alphabetical?

Reply:
@vkarpov15 This is as per Mongo's own definitions from https://www.mongodb.com/docs/v6.0/reference/operator/update/#std-label-update-operators-processing-order

"Starting in MongoDB 5.0, update operators process document fields with string-based names in lexicographic order. "

I agree that if it was up to me, I'd have strictly followed order in which operations are defined in message -- that seems more like caller's intent.

Reply:
That's fair, I wasn't aware of MongoDB's ordering. I guess in most cases this ordering doesn't matter, because document updates are atomic in MongoDB and you can't update the same property using 2 different update operators. You can disregard my comment.

Reply:
Right, I just happened to notice it (based on Aaron's comments). And I agree that in almost all cases it is irrelevant -- the main case I see is the insertion order of fields that depends on ordering.